### PR TITLE
Remove default props from functional components

### LINKED ISSUE #1905: Warning in the console when Using DefaultProps on Function Components
## Description
defaultProps on function components has been deprecated by React (see: https://github.com/facebook/react/pull/16210), therefore there are lots of warning in our console right now

Screenshot 2024-05-15 at 11 08 09

Todo: to remove defaultProps in our components

```
-- If you received an error, place it here.
```

```
-- Separate them if you have more than one.
```

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version:
* Operating system and version:
* NodeJS version used (if relevant):
* Kedro version used (if relevant):
* Python version used (if relevant):

## Checklist
- [ ] Include labels so that we can categorise your issue

### PR DESCRIPTION
**Pull request by Huongg**
## Description
Fixes #1905 

## Development notes
I've removed the defaults props from our Functional components, set them directly in the props values instead

## QA notes
All the tests have passed and everything seems to work fine form my side

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/components/ui/dropdown/dropdown-renderer.js [hunk 1]
```diff
@@ -169,23 +169,6 @@ const DropdownRenderer = ({
   );
 };
 
-DropdownRenderer.defaultProps = {
-  children: null,
-  defaultText: 'Please select...',
-  disabled: false,
```
Comment (tynandebold) [inline src/components/ui/dropdown/dropdown-renderer.js:175]: You might want to add the `false / true` values as defaults in the props?

**Review by tynandebold — APPROVED** [review]
Looks great! I left one comment.

I did a project-wide find for `defaultProps` and found one left: https://github.com/kedro-org/kedro-viz/blob/63cfeaa6e11b32b40052830e49ba267cf2ed9010/tools/test-lib/react-app/app.js#L80

But it's in the `/tools/test-lib` directory so I don't think it matters too much.

File: src/components/ui/dropdown/dropdown-renderer.js (hunk 1 above)
Comment (Huongg) [reply src/components/ui/dropdown/dropdown-renderer.js:175]: ah yeah true 👍, dont know why the tests didnt catch this

**Comment by Huongg** [comment]
> Looks great! I left one comment.
> 
> I did a project-wide find for `defaultProps` and found one left:
> 
> https://github.com/kedro-org/kedro-viz/blob/63cfeaa6e11b32b40052830e49ba267cf2ed9010/tools/test-lib/react-app/app.js#L80
> 
> But it's in the `/tools/test-lib` directory so I don't think it matters too much.

yeah I didn't remove it cause it seems like there was no error, but maybe best to remove it too, to be consistent with the rest

**Review by rashidakanchwala — APPROVED** [review]

**Pull request merged**